aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -67,11 +69,19 @@ module aptos_framework::object {
     /// A shared resource group for storing object resources together in storage.
     struct ObjectGroup { }
 
-    /// Type safe way of designate an object as at this address.
+    /// Type-safe way of designate an object as at this address.
     struct ObjectId has copy, drop, store {
         inner: address,
     }
 
+    /// A typed ObjectId -- these can only provide guarantees based upon the underlying data type,
+    /// that is the validity of T existing at an address is something that cannot be verified by
+    /// any other module than the module that defined T. Similarly, the module that defines T can
+    /// remove it from storage at any point in time.
+    struct TypedObjectId<phantom T> has copy, drop, store {
```

Comment:
This name is not nice. "typed" is not something which should be emphasized in the very much typed Move language. Perhaps ObjectId should be RawObjectId? Or use `Ref<T>`?